Library/Homebrew/formula.rb

```diff
@@ -2842,7 +2841,7 @@ def eligible_kegs_for_cleanup(quiet: false)
       eligible_kegs = if head? && (head_prefix = latest_head_prefix)
         head, stable = installed_kegs.partition { |k| k.version.head? }
         # Remove newest head and stable kegs
-        head - [Keg.new(head_prefix)] + stable.sort_by(&:version).slice(0...-1)
+        head - [Keg.new(head_prefix)] + Keg.sort(stable).drop(1)
```

Comment:
I think this logic is different from before since we're now doing a reverse sort by version. Before we were dropping the newest version and now we're dropping the oldest one.

Reply:
`slice(0...-1)` drops the last entry but `drop(1)` drops the first entry